capstone/src/main/java/com/google/sps/servlets/QuadTreeServlet.java

```diff
@@ -0,0 +1,158 @@
+import java.util.Scanner;
+import java.util.Random;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.HashSet;
+import com.google.gson.Gson;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import java.io.PrintWriter;
+import com.google.sps.Objects.QuadTree;
+import com.google.sps.Objects.Coordinate;
+import com.google.sps.Objects.GroupLocation;
+import com.google.sps.Objects.Location;
+import com.google.sps.Objects.BoundingBox;
+import error.ErrorHandler;
+
+/*
+ * Parses a TSV file of fake group data and populates the Datastore with Group and Post entities.
+*/
+@WebServlet("/create-quadtree")
+public class QuadTreeServlet extends HttpServlet {
+
+  private QuadTree quadTree;
+  private final String QUADTREE_FILE = "../../data/quadtree";
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+    this.getQuadTreeFromFile();
+  }
+
+  @Override
+  public void destroy() {
+    this.saveState();
+  }
+
+  private void getQuadTreeFromFile() {
+    try {
+      FileInputStream fileInput = new FileInputStream(new File(QUADTREE_FILE));
+      ObjectInputStream objectInput = new ObjectInputStream(fileInput);
+      quadTree = (QuadTree) objectInput.readObject();
+
+      fileInput.close();
+      objectInput.close();
+      return;
+    } catch (FileNotFoundException e) {
+      System.err.println("File does not exist");
+    } catch (IOException e) {
+      System.err.println("Cannot read from file");
+    } catch (ClassNotFoundException e) {
+      System.err.println("Class not found");
+    }
+
+    quadTree = new QuadTree(
+        /*NODE_CAPACITY*/ 4, 
+        /*level*/ 0, 
+        /*BoudingBox*/ new BoundingBox(25.641526, -130.429688, 50.847573, -66.269531));
+    populateQuadTree(quadTree);
+    saveState();
+  }
+
+  private void populateQuadTree(QuadTree quadTree) {
+    Scanner scanner = new Scanner(
+      getServletContext().getResourceAsStream("/WEB-INF/fast-food-location.csv"));
```

Comment:
Can you add the CSV name as a constant? Will make this less dependent on this dataset and allow you to swap datasets if you wanted to